internal/fwserver/write_only_nullification.go

```diff
@@ -58,14 +57,6 @@ func NullifyWriteOnlyAttributes(ctx context.Context, resourceSchema fwschema.Sch
 		// Value type from new state to create null with
 		newValueType := attribute.GetType().TerraformType(ctx)
 
-		// If the attribute is dynamic set the new value type to DynamicPseudoType
```

Comment:
Addresses: https://github.com/hashicorp/terraform-plugin-framework/pull/1044#discussion_r1946625602

We've verified with Core that erasing the underlying type is fine for write-only value nullification, so this conditional is no longer needed.